### PR TITLE
update pplnn to 0.9.1

### PR DESCRIPTION
**Pull request by ouonline**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

ppl.nn 0.9.0 has some minor changes on c++ api.

## Modification

update docs about supported version of ppl.nn and remove deprecated calls.

## BC-breaking (Optional)

releases before 0.9.0 are not supported after this pr

## Use cases (Optional)

none

### CODE REVIEW

**Comment by lzhangzz** [comment]
@ouonline 

Using ppl.nn together with ppl.cv failed due to ppl.cv depends on older versions of hpcc and ppl.common.
```
/usr/bin/ld: /workspace/ppl.nn/pplnn-build/install/lib/libpplcommon_static.a(log.cc.o): in function `ppl::common::LogMessage::~LogMessage()':
/workspace/ppl.nn/deps/pplcommon/src/ppl/common/log.cc:62: multiple definition of `ppl::common::LogMessage::~LogMessage()'; /workspace/ppl.cv/cuda-build/install/lib/cmake/ppl/../../../lib/libpplcommon_static.a(log.cc.o):log.cc:(.text._ZN3ppl6common10LogMessageD2Ev+0x0): first defined here
/usr/bin/ld: /workspace/ppl.nn/pplnn-build/install/lib/libpplcommon_static.a(log.cc.o): in function `ppl::common::LogMessage::~LogMessage()':
/workspace/ppl.nn/deps/pplcommon/src/ppl/common/log.cc:62: multiple definition of `ppl::common::LogMessage::~LogMessage()'; /workspace/ppl.cv/cuda-build/install/lib/cmake/ppl/../../../lib/libpplcommon_static.a(log.cc.o):log.cc:(.text._ZN3ppl6common10LogMessageD2Ev+0x0): first defined here
/usr/bin/ld: /workspace/ppl.nn/pplnn-build/install/lib/libpplcommon_static.a(log.cc.o): in function `ppl::common::LogMessage::operator<<(void const*)':
log.cc:(.text._ZN3ppl6common10LogMessagelsEPKv+0x0): multiple definition of `ppl::common::LogMessage::operator<<(void const*)'; /workspace/ppl.cv/cuda-build/install/lib/cmake/ppl/../../../lib/libpplcommon_static.a(log.cc.o):log.cc:(.text._ZN3ppl6common10LogMessagelsEPKv+0x0): first defined here
/usr/bin/ld: /workspace/ppl.nn/pplnn-build/install/lib/libpplcommon_static.a(log.cc.o): in function `ppl::common::SetCurrentLogger(ppl::common::Logger*)':
/workspace/ppl.nn/deps/pplcommon/src/ppl/common/log.cc:133: multiple definition of `ppl::common::SetCurrentLogger(ppl::common::Logger*)'; /workspace/ppl.cv/cuda-build/install/lib/cmake/ppl/../../../lib/libpplcommon_static.a(log.cc.o):log.cc:(.text._ZN3ppl6common16SetCurrentLoggerEPNS0_6LoggerE+0x0): first defined here
/usr/bin/ld: /workspace/ppl.nn/pplnn-build/install/lib/libpplcommon_static.a(log.cc.o): in function `ppl::common::GetCurrentLogger()':
/workspace/ppl.nn/deps/pplcommon/src/ppl/common/log.cc:137: multiple definition of `ppl::common::GetCurrentLogger()'; /workspace/ppl.cv/cuda-build/install/lib/cmake/ppl/../../../lib/libpplcommon_static.a(log.cc.o):log.cc:(.text._ZN3ppl6common16GetCurrentLoggerEv+0x0): first defined here
```
Also, build script `tools/scripts/build_ubuntu_x64_pplnn.py` need to be updated as well.

**Comment by ouonline** [comment]
I've updated ppl.cv's deps to be the same as what ppl.nn uses. and the build scrips is also updated.

**Review by lzhangzz — APPROVED** [review]
LGTM

**Comment by lzhangzz** [comment]
@ouonline After upgrading to v0.9.0, DBNet model accuracy drops to zero, it was OK when using v0.8.2.

Here is how the model is converted.
```
python tools/deploy.py \
        --device cuda \
        --dump-info \
        --work-dir textdet_tmp_model \
        $MMOCR/text-detection/text-detection_pplnn_dynamic-640x640.py \
        $MMOCR/configs/textdet/dbnet/dbnet_r18_fpnc_1200e_icdar2015.py \
        $MMOCR/models/dbnet_r18_fpnc_sbn_1200e_icdar2015_20210329-ba3ab597.pth \
        $MMOCR/demo/demo_text_ocr.jpg
```

**Comment by jianfei-wangg** [comment]
> @ouonline After upgrading to v0.9.0, DBNet model accuracy drops to zero, it was OK when using v0.8.2.
> 
> Here is how the model is converted.
> 
> ```
> python tools/deploy.py \
>         --device cuda \
>         --dump-info \
>         --work-dir textdet_tmp_model \
>         $MMOCR/text-detection/text-detection_pplnn_dynamic-640x640.py \
>         $MMOCR/configs/textdet/dbnet/dbnet_r18_fpnc_1200e_icdar2015.py \
>         $MMOCR/models/dbnet_r18_fpnc_sbn_1200e_icdar2015_20210329-ba3ab597.pth \
>         $MMOCR/demo/demo_text_ocr.jpg
> ```

Hello, I'm tring to convert DBNet, can you provide the related file "text-detection_pplnn_dynamic-640x640.py" or the file url, thanks.

**Comment by irexyc** [comment]
> Hello, I'm tring to convert DBNet, can you provide the related file "text-detection_pplnn_dynamic-640x640.py" or the file url, thanks.

Hi, @jianfei-wangg 
I think the config file is this url https://github.com/open-mmlab/mmdeploy/blob/master/configs/mmocr/text-detection/text-detection_pplnn_dynamic-640x640.py

**Comment by lvhan028** [comment]
> > @ouonline After upgrading to v0.9.0, DBNet model accuracy drops to zero, it was OK when using v0.8.2.
> > Here is how the model is converted.
> > ```
> > python tools/deploy.py \
> >         --device cuda \
> >         --dump-info \
> >         --work-dir textdet_tmp_model \
> >         $MMOCR/text-detection/text-detection_pplnn_dynamic-640x640.py \
> >         $MMOCR/configs/textdet/dbnet/dbnet_r18_fpnc_1200e_icdar2015.py \
> >         $MMOCR/models/dbnet_r18_fpnc_sbn_1200e_icdar2015_20210329-ba3ab597.pth \
> >         $MMOCR/demo/demo_text_ocr.jpg
> > ```
> 
> Hello, I'm tring to convert DBNet, can you provide the related file "text-detection_pplnn_dynamic-640x640.py" or the file url, thanks.

Hi, is there any update? @jianfei-wangg

**Comment by ouonline** [comment]
> > > @ouonline After upgrading to v0.9.0, DBNet model accuracy drops to zero, it was OK when using v0.8.2.
> > > Here is how the model is converted.
> > > ```
> > > python tools/deploy.py \
> > >         --device cuda \
> > >         --dump-info \
> > >         --work-dir textdet_tmp_model \
> > >         $MMOCR/text-detection/text-detection_pplnn_dynamic-640x640.py \
> > >         $MMOCR/configs/textdet/dbnet/dbnet_r18_fpnc_1200e_icdar2015.py \
> > >         $MMOCR/models/dbnet_r18_fpnc_sbn_1200e_icdar2015_20210329-ba3ab597.pth \
> > >         $MMOCR/demo/demo_text_ocr.jpg
> > > ```
> > 
> > 
> > Hello, I'm tring to convert DBNet, can you provide the related file "text-detection_pplnn_dynamic-640x640.py" or the file url, thanks.
> 
> Hi, is there any update? @jianfei-wangg

problem fixed in pplnn v0.9.1. latest changes are force pushed.

File: docs/en/05-supported-backends/pplnn.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # PPLNN Support
 
-MMDeploy supports ppl.nn v0.8.1 and later. This tutorial is based on Linux systems like Ubuntu-18.04.
+MMDeploy supports ppl.nn v0.9.0 and later. This tutorial is based on Linux systems like Ubuntu-18.04.
```
Comment (lvhan028) [inline docs/en/05-supported-backends/pplnn.md:3]: May change `v0.9.1`

File: docs/zh_cn/05-supported-backends/pplnn.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # PPLNN 支持情况
 
-MMDeploy supports ppl.nn v0.8.1 and later. This tutorial is based on Linux systems like Ubuntu-18.04.
+MMDeploy supports ppl.nn v0.9.0 and later. This tutorial is based on Linux systems like Ubuntu-18.04.
```
Comment (lvhan028) [inline docs/zh_cn/05-supported-backends/pplnn.md:3]: May change `v0.9.1`

File: docs/en/05-supported-backends/pplnn.md (hunk 1 above)
Comment (ouonline) [reply docs/en/05-supported-backends/pplnn.md]: done

File: docs/zh_cn/05-supported-backends/pplnn.md (hunk 2 above)
Comment (ouonline) [reply docs/zh_cn/05-supported-backends/pplnn.md]: done

**Comment by lzhangzz** [comment]
@ouonline Getting the following error when running on a GeForce GTX 1660 Ti using either v0.9.0 or v0.9.1. I was testing on an 2080Ti before, but I don't have access to it anymore.
```
[ERROR][2022-12-12 12:26:57.781][reciprocal_kernel.cc:33] Reciprocal op only support float16 and float32 for now.
[ERROR][2022-12-12 12:26:57.781][sequential_scheduler.cc:129] exec kernel[Reciprocal_125] of type[:Reciprocal:11] failed: invalid value
[ERROR][2022-12-12 12:26:57.781][runtime_impl.cc:337] Run() failed: invalid value
```

**Comment by jianfei-wangg** [comment]
> @ouonline Getting the following error when running on a GeForce GTX 1660 Ti using either v0.9.0 or v0.9.1. I was testing on an 2080Ti before, but I don't have access to it anymore.
> 
> ```
> [ERROR][2022-12-12 12:26:57.781][reciprocal_kernel.cc:33] Reciprocal op only support float16 and float32 for now.
> [ERROR][2022-12-12 12:26:57.781][sequential_scheduler.cc:129] exec kernel[Reciprocal_125] of type[:Reciprocal:11] failed: invalid value
> [ERROR][2022-12-12 12:26:57.781][runtime_impl.cc:337] Run() failed: invalid value
> ```

hello, which model did you test and can you provide the related commands ?

**Comment by lzhangzz** [comment]
> hello, which model did you test and can you provide the related commands ? @lzhangzz

the same DBNet model mentioned above.

**Comment by jianfei-wangg** [comment]
> Reciprocal o

@lzhangzz I just explored the generated dbnet onnx model, but no Reciprocal ops are found. I use the onnx-1.12.0, mmdeploy-0.11.0, pytorch-1.8.0, the main tree of mmocr and the mater tree of mmdelopy, any difference between us?

**Comment by ouonline** [comment]
> @jianfei-wangg I'm using torch-1.12. Starting from torch-1.11 the exported ONNX for `torch.reciprocal` has become `Reciprocal` instead of something like `Div(1, x)`.

problem solved. I've re-tagged the latest version to be v0.9.1. please try again.

**Comment by lzhangzz** [comment]
@ouonline Thanks, everything works now! However, the inference speed is quite slow compared with TensorRT or even ONNX Runtime.
| FPS@batch=1 | ppl.nn  | TensorRT | ONNX Runtime |
|-------------|--------|----------|--------------|
| DBNet-r18   | 5.40   | 31.36    | 26.67        |
| YOLOX-s     | 18.06  | 67.41    | 37.89        |
| ResNet-18   | 145.34 | 440.28   | 435.37       |

is this expected due to the lack of hardware WMMA on 1660Ti?

**Review by lvhan028 — APPROVED** [review]

**Comment by OpenMMLab-Miao** [comment]
Dear ouonline ,
First of all, we want to express our gratitude for your significant PR in the MMDeploy project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.
If you are Chinese or have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. We look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG
Thank you again for your contribution❤
Best regards！ @ouonline

**Pull request merged**